sui_programmability/examples/games_with_chance/tests/test_satoshi_flip.move

```diff
@@ -0,0 +1,724 @@
+        let house = @0xBAE;
+        let player = @0xFAB;
+        let secret = b"supersecret";
+        let secret_hash = sha3_256(secret);
+        let min_bet = 100;
+        let max_bet = 5000;
+
+        let scenario_val = test_scenario::begin(world);
+        let scenario = &mut scenario_val;
+        {
+            start(test_scenario::ctx(scenario), house, player);
+        };
+
+        // house creates the game
+        test_scenario::next_tx(scenario, house);
+        {
+            let coinA = test_scenario::take_from_sender<Coin<SUI>>(scenario);
+            let ctx = test_scenario::ctx(scenario);
+            satoshi_flip::start_game(secret_hash, coinA, max_bet, min_bet, ctx);
+        };
+
+        // house ends game
+        test_scenario::next_tx(scenario, house);
+        {
+            let game_val = test_scenario::take_shared<Game>(scenario);
+            let game = &mut game_val;
+            let ctx = test_scenario::ctx(scenario);
+
+            satoshi_flip::end_game(game, secret, ctx);
+
+            test_scenario::return_shared(game_val);
+        };
+
+        // check house's balance
+        test_scenario::next_tx(scenario, house);
+        {
+            // check that house has the correct amount
+            let coin = test_scenario::take_from_sender<Coin<SUI>>(scenario);
+            assert!(coin::value(&coin) == 50000, EWronghouseTotal);
+            test_scenario::return_to_sender(scenario, coin);
+        };
+
+        test_scenario::end(scenario_val);
+    }
+
+
+    // Tests expecting abort
+
+    // tests for start_game with wrong inputs
+    // Check that min_bet <= max_bet is enforced properly
+    #[test]
+    #[expected_failure(abort_code = 4)]
+    fun house_wrong_min_max_bet() {
+        let world = @0x1EE7; // needed only for beginning the test_scenario
+        let house = @0xBAE;
+        let player = @0xFAB;
+        let secret = b"supersecret";
+        let secret_hash = sha3_256(secret);
+        let max_bet = 5000;
+        let min_bet = 10000; // this is too high here
+
+        let scenario_val = test_scenario::begin(world);
+        let scenario = &mut scenario_val;
+        {
+            start(test_scenario::ctx(scenario), house, player);
+        };
+
+        // house creates the game
+        test_scenario::next_tx(scenario, house);
+        {
+            let coinA = test_scenario::take_from_sender<Coin<SUI>>(scenario);
+            let ctx = test_scenario::ctx(scenario);
+            satoshi_flip::start_game(secret_hash, coinA, max_bet, min_bet, ctx);
+        };
+        test_scenario::end(scenario_val);
+    }
+
+    // Check that house provided coin of sufficient amount to cover the max_bet
+    #[test]
+    #[expected_failure(abort_code = 9)]
+    fun house_insufficient_balance() {
+        let world = @0x1EE7; // needed only for beginning the test_scenario
+        let house = @0xBAE;
+        let player = @0xFAB;
+        let secret = b"supersecret";
```

Comment:
Note that ideally we want to recommend secrets being unpredictable random bytes of size at least 16 bytes. Please add this statement to the docs as well.